File: src/Conduit/Models/Article.php
<?php

namespace Conduit\Models;

use Illuminate\Database\Eloquent\Model;


/**
 * @property integer        id
 * @property string         slug
 * @property string         title
 * @property string         description
 * @property string         body
 * @property integer        user_id
 * @property \Carbon\Carbon created_at
 * @property \Carbon\Carbon update_at
 */
class Article extends Model
{

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'slug',
        'title',
        'description',
        'body',
        'user_id',
    ];

    public function setSlugAttribute($value)
    {
        $index = 0;
        $slug = $value;
        while (self::newQuery()
            ->where('slug', $slug)
            ->where('id', '!=', $this->id)
            ->exists()) {
            $slug = $value . '-' . ++$index;
        }

        return $this->attributes['slug'] = $slug;
    }

    /********************
     *  Relationships
     ********************/

    public function tags()
    {
        return $this->belongsToMany(Tag::class);
    }
}

File: app/models/article.go
package models

import (
	"gopkg.in/gorp.v2"
	"time"
)

type Article struct {
	ID             int
	Slug           string
	Title          string
	Description    string
	Body           string
	UserID         int
	FavoritesCount int
	CreatedAt      time.Time
	UpdatedAt      time.Time

	// Transient
	User User
}

func (i *Article) PreInsert(s gorp.SqlExecutor) error {
	i.CreatedAt = time.Now()
	i.UpdatedAt = i.CreatedAt
	return nil
}

func (i *Article) PreUpdate(s gorp.SqlExecutor) error {
	i.UpdatedAt = time.Now()
	return nil
}
